sklearnex/_utils.py

```diff
@@ -63,3 +63,69 @@ def get_patch_message(s):
 def get_sklearnex_version(rule):
     from daal4py.sklearn._utils import daal_check_version
     return daal_check_version(rule)
+
+
+def get_patch_str(name=None, verbose=True):
+    return f"""try:
+    from sklearnex import patch_sklearn
+    patch_sklearn(name={str(name)}, verbose={str(verbose)})
+    del patch_sklearn
+except (ImportError, ModuleNotFoundError):
+    pass"""
+
+
+def get_patch_str_re():
+    return r"""\ntry:
+    from sklearnex import patch_sklearn
+    patch_sklearn\(name=.*, verbose=.*\)
+    del patch_sklearn
+except \(ImportError, ModuleNotFoundError\):
+    pass\n"""
+
+
+def patch_sklearn_global(name=None, verbose=True):
+    import os
+    import re
+    try:
+        import sklearn
+    except ImportError:
+        raise ImportError("Scikit-learn could not be imported. Nothing to patch\n")
+
+    init_file_path = sklearn.__file__
+    distributor_file_path = os.path.join(os.path.dirname(init_file_path),
+                                         "_distributor_init.py")
+
+    with open(distributor_file_path, 'r', encoding='utf-8') as distributor_file:
+        lines = distributor_file.read()
+        if re.search(get_patch_str_re(), lines):
+            lines = re.sub(get_patch_str_re(), '', lines)
+
+    with open(distributor_file_path, 'w', encoding='utf-8') as distributor_file:

```

Comment:
Does this mean it will always overwrite?

Reply:
Yes, we overwrite file with patching string